test/functional/feature_proxy.py

```diff
@@ -336,20 +336,29 @@ def networks_dict(d):
         msg = "Error: Invalid -i2psam address or hostname: 'def:xyz'"
         self.nodes[1].assert_start_raises_init_error(expected_msg=msg)
 
+        self.log.info("Test passing -onlynet=onion with -onion=0/-noonion raises expected init error")
         msg = (
             "Error: Outbound connections restricted to Tor (-onlynet=onion) but "
-            "the proxy for reaching the Tor network is not provided (no -proxy= "
-            "and no -onion= given) or it is explicitly forbidden (-onion=0)"
+            "the proxy for reaching the Tor network is explicitly forbidden: -onion=0"
         )
-        self.log.info("Test passing -onlynet=onion without -proxy or -onion raises expected init error")
-        self.nodes[1].extra_args = ["-onlynet=onion"]
-        self.nodes[1].assert_start_raises_init_error(expected_msg=msg)
-
-        self.log.info("Test passing -onlynet=onion with -onion=0/-noonion raises expected init error")
         for arg in ["-onion=0", "-noonion"]:
             self.nodes[1].extra_args = ["-onlynet=onion", arg]
             self.nodes[1].assert_start_raises_init_error(expected_msg=msg)
 
+        self.log.info("Test passing -onlynet=onion without -proxy, -onion or -listenonion raises expected init error")
+        self.nodes[1].extra_args = ["-onlynet=onion", "-listenonion=0"]
+        msg = (
+            "Error: The -onlynet=onion configuration option was passed to restrict outbound"
+            " connections to Tor, but the proxy for reaching the Tor network was not"
+            " provided: none of -proxy, -onion or -listenonion was given"
+        )
+        self.nodes[1].assert_start_raises_init_error(expected_msg=msg)
+
+        self.log.info("Test passing -onlynet=onion without -proxy or -onion but with -listenonion=1 is ok")
+        self.nodes[1].extra_args = ["-onlynet=onion", "-listenonion=1"]
+        self.start_node(1)
```

Comment:
- these two lines can be combined to one, i.e. `self.start_node(1, extra_args = ["-onlynet=onion", "-listenonion=1"])`
- maybe check that Tor is successfully up with assert_debug_log

```diff
         self.log.info("Test passing -onlynet=onion without -proxy or -onion but with -listenonion=1 is ok")
-        self.nodes[1].extra_args = ["-onlynet=onion", "-listenonion=1"]
-        self.start_node(1)
+        with self.nodes[1].assert_debug_log(["[tor] Authentication successful"]):
+            self.start_node(1, extra_args = ["-onlynet=onion", "-listenonion=1"])
         self.stop_node(1)
```

Reply:
I combined the two lines to one, as suggested.

However, expecting "[tor] Authentication successful" in the log failed for me because it tried to talk to a Tor daemon on `127.0.0.1:9051`, not to the Python one from the testing framework at `self.conf2.addr`, because when overriding `extra_args` the setting `-onion=...` which was in `extra_args` is lost.